csharp/src/Apache.Arrow/ArrowBuffer.Builder.cs

```diff
@@ -91,13 +88,8 @@ public Builder<T> Reserve(int capacity)
 
             public Builder<T> Resize(int capacity)
             {
-                if (capacity < 0)
-                {
-                    throw new ArgumentOutOfRangeException(nameof(capacity));
-                }
-
-                Reallocate(capacity);
-                Length = Math.Min(Length, capacity);
+                EnsureCapacity(capacity);
```

Comment:
There's still a bug here if capacity < Length, but I'm open to fixing it in a different PR

Reply:
Is the bug simply that we will allocate more memory than necessary? Functionally it will be correct though, right?